test_conformance/extensions/cl_khr_command_buffer/command_buffer_test_copy.cpp

```diff
@@ -0,0 +1,459 @@
+                origin,
+                region,
+                0,
+                nullptr,
+                nullptr,
+                nullptr);
+
+            test_error(error, "clCommandFillImageKHR failed");
+
+            error = clCommandCopyImageKHR(
+                    command_buffer,
+                    nullptr,
+                    src_image,
+                    dst_image,
+                    origin,
+                    origin,
+                    region,
+                    0,
+                    0,
+                    nullptr,
+                    nullptr);
+
+            test_error(error, "clCommandCopyImageKHR failed");
+
+            error = clFinalizeCommandBufferKHR(command_buffer);
+            test_error(error, "clFinalizeCommandBufferKHR failed");
+
+            error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                            nullptr, nullptr);
+            test_error(error, "clEnqueueCommandBufferKHR failed");
+
+            error = clFinish(queue);
+
+            test_error(error, "clFinish failed");
+
+            std::vector<cl_char> output_data(data_size);
+            error = clEnqueueReadImage(queue, dst_image, CL_TRUE, origin, region, 0, 0, output_data.data(), 0, nullptr, nullptr);
+
+            for (size_t i = 0; i < data_size; i++)
+            {
+                CHECK_VERIFICATION_ERROR(pattern, output_data[i], i);
+            }
+
+            return CL_SUCCESS;
+        }
+
+        cl_int SetUp(int elements) override
+        {
+            cl_int error = BasicCommandBufferTest::SetUp(elements);
+            test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+            src_image = create_image_2d(context, CL_MEM_READ_WRITE, &formats, img_width,
+                            img_height, 0, NULL, &error);
+            test_error(error, "create_image_2d failed");
+
+            dst_image = create_image_2d(context, CL_MEM_READ_WRITE, &formats, img_width,
+                            img_height, 0, NULL, &error);
+            test_error(error, "create_image_2d failed");
+
+            return CL_SUCCESS;
+        }
+
+        int        img_width = 512;
+        int        img_height = 512;
+        const size_t data_size = img_width * img_height * 4;
+        const size_t origin[3]={0,0,0}, region[3]={img_width,img_height,1};
+        const char pattern = 0x05;
+        const cl_uint fill_color[4] = { pattern, pattern, pattern, pattern };
+        const cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+        clMemWrapper src_image;
+        clMemWrapper dst_image;
+    };
+
+    struct CopyBufferKHR : public BasicCommandBufferTest
+    {
+        using BasicCommandBufferTest::BasicCommandBufferTest;
+
+        cl_int Run() override
+        {
+            cl_int error = clCommandFillBufferKHR(
+                    command_buffer, nullptr, in_mem, &pattern, sizeof(cl_char), 0,
+                    data_size, 0, nullptr, nullptr, nullptr);
```

Comment:
Using hard-coded `data_size` without involving `sizeof` and using the same name as parent method name is a bit misleading I think.

Reply:
Done